CONTRIBUTING.md

```diff
@@ -1,159 +1,307 @@
 
-* You will need a [GitHub account](https://github.com/signup/free).
-* Submit a [ticket for your issue](https://github.com/coreruleset/coreruleset/issues), assuming one does not already exist.
-  * Clearly describe the issue including steps to reproduce when it is a bug.
-  * Make sure you specify the version that you know has the issue.
-  * Bonus points for submitting a failing test along with the ticket.
-* If you don't have push access, fork the repository on GitHub.
+## Getting Started with a New Contribution
+
+1. Sign in to [GitHub](https://github.com/join).
+2. Open a [new issue](https://github.com/coreruleset/coreruleset/issues) for the contribution, *assuming a similar issue doesn't already exist*.
+    * **Clearly describe the issue**, including steps to reproduce if reporting a bug.
+    * **Specify the CRS version in question** if reporting a bug.
+    * Bonus points for submitting tests along with the issue.
+3. Fork the repository on GitHub and begin making changes there.
+4. Signed commits are preferred. (For more information and help with this, refer to the [GitHub documentation](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits)).
 
 ## Making Changes
 
-* Please base your changes on the latest dev branch (e.g. ```v3.4/dev```)
-* Create a topic branch for your feature or bug fix.
-* Please fix only one problem at a time; this will help to quickly test and merge your change. If you intend to fix multiple unrelated problems, please use a separate branch for each problem.
+* Base any changes on the latest dev branch (e.g., `v3.4/dev`).
+* Create a topic branch for each new contribution.
+* Fix only one problem at a time. This helps to quickly test and merge submitted changes. If intending to fix *multiple unrelated problems* then use a separate branch for each problem.
 * Make commits of logical units.
-* Make sure your commits adhere to the rules guidelines below.
-* Make sure your commit messages are in the [proper format](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html): The first line of the message should have 50 characters or less, separated by a blank line from the (optional) body. The body should be wrapped at 70 characters and paragraphs separated by blank lines. Bulleted lists are also fine.
+* Make sure commits adhere to the contribution guidelines presented in this document.
+* Make sure commit messages follow the [standard Git format](http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
+
+## General Formatting Guidelines for Rules Contributions
+
+* American English should be used throughout.
+* 4 spaces should be used for indentation (no tabs).
+* No trailing whitespace at EOL.
+* No trailing blank lines at EOF.
```

Comment:
Ah, is that a new thing with the new linter? I think it's said "no trailing lines at EOF" for years.

So, replace with "Must have a trailing line at EOF"?

@fzipi Git blame says you added that line in originally :P Does this sound right?

Reply:
I just checked the linter configuration and it requires a newline as the last character before EOF.

Reply:
Thanks for checking. So, in that case, the instructions are correct as they currently are, right?

Reply:
Maybe it's just the wording but "no trailing blank lines" to me means that the last character at EOF must _not_ be a newline.

The last point makes the second to last obsolete, doesn't it? I would keep the last point and remove the second to last.

Reply:
Hmmm.... "no trailing blank lines" is supposed to cover multiple empty lines at EOF. If we remove it, the other two won't cover that one, right? We need it.

Reply:
Right. "Files must end in a single newline character". That should do it.

Reply:
Great. Unless there are any objections, I'm going with the following:

- Files must end with a single newline character.
- No trailing whitespace at EOL.
- No trailing blank lines at EOF (only the required single EOF newline character is allowed).

I added the explanation in brackets because otherwise "Files must end with a single newline character" and "No trailing blank lines at EOF" seem contradictory.

Reply:
Perfect.